lib/ansible/playbook/base.py

```diff
@@ -315,7 +316,10 @@ def copy(self):
         Create a copy of this object and return it.
         '''
 
-        new_me = self.__class__()
+        try:
+            new_me = self.__class__()
+        except RuntimeError:
+            raise AnsibleError("Exceeded maximum object depth. This may have been caused by excessive role recursion.")
```

Comment:
Try using `six.reraise()` here so that under py3 it'd show the previous exception trace